File: 27_MirrorOfBinaryTree/MirrorOfBinaryTree.cpp
struct TreeNode
{
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};

class Solution
{
  public:
    void Mirror(TreeNode *pRoot)
    {
        if (pRoot == nullptr)
            return;
        if (pRoot->left == nullptr && pRoot->right == nullptr)
            return;
        TreeNode *pTemp = pRoot->left;
        pRoot->left = pRoot->right;
        pRoot->right = pTemp;
        Mirror(pRoot->left);
        Mirror(pRoot->right);
    }
};

File: 27_MirrorOfBinaryTree/MirrorOfBinaryTree.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    # 返回镜像树的根节点
    def Mirror(self, root):
        # write code here
        if root is None:
            return
        if root.left is None and root.right is None:
            return
        root.left, root.right = root.right, root.left
        self.Mirror(root.left)
        self.Mirror(root.right)